src/Mod/Sketcher/Gui/DrawSketchHandler.cpp

```diff
@@ -420,137 +420,131 @@ DrawSketchHandler::suggestedConstraintsPixmaps(std::vector<AutoConstraint>& sugg
-                if (geo->is<Part::GeomLineSegment>()) {
-                    auto* line = static_cast<const Part::GeomLineSegment*>(geo);
-
-                    Base::Vector2d startPoint = toVector2d(line->getStartPoint());
-                    Base::Vector2d endPoint = toVector2d(line->getEndPoint());
-                    Base::Vector2d midPoint = (startPoint + endPoint) / 2;
-
-                    // Check if we are at middle of the line
-                    if ((Pos - midPoint).Length() < (endPoint - startPoint).Length() * 0.05) {
-                        lineCenter = true;
-                    }
-                }
-
+            if (preSel.posId == PointPos::none) {
+                bool lineCenter = isLineCenterAutoConstraint(preSel.geoId, Pos);
                 constr.Type = lineCenter ? Sketcher::Symmetric : Sketcher::PointOnObject;
             }
             else {
                 constr.Type = Sketcher::Coincident;
             }
         }
-        else if (type == AutoConstraint::CURVE && PosId != PointPos::none) {
+        else if (type == AutoConstraint::CURVE && preSel.posId != PointPos::none) {
             constr.Type = Sketcher::PointOnObject;
         }
-        else if (type == AutoConstraint::CURVE && PosId == PointPos::none) {
+        else if (type == AutoConstraint::CURVE && preSel.posId == PointPos::none) {
             constr.Type = Sketcher::Tangent;
         }
 
-        if (constr.Type == Sketcher::Tangent && preselectIsLine) {
-            if (Dir.Length() < 1e-8 || hitShapeDir.Length() < 1e-8) {
-                // Direction not set so return;
-                return suggestedConstraints.size();
+        if (constr.Type == Sketcher::Tangent && preSel.isLine) {
+            if (Dir.Length() < 1e-8 || preSel.hitShapeDir.Length() < 1e-8) {
+                return;  // Direction not set so return;
             }
+
             // We are hitting a line and have hitting vector information
             Base::Vector3d dir3d = Base::Vector3d(Dir.x, Dir.y, 0);
-            double cosangle = dir3d.Normalize() * hitShapeDir.Normalize();
+            double cosangle = dir3d.Normalize() * preSel.hitShapeDir.Normalize();
 
             // the angle between the line and the hitting direction are over around 6 degrees (it is
             // substantially parallel) or if it is an sketch axis (that can not move to accommodate
             // to the shape), then only if it is around 6 degrees with the normal (around 84
             // degrees)
-            if (fabs(cosangle) < 0.995f
-                || ((GeoId == Sketcher::GeoEnum::HAxis || GeoId == Sketcher::GeoEnum::VAxis)
-                    && fabs(cosangle) < 0.1)) {
-                suggestedConstraints.push_back(constr);
+            if (fabs(cosangle) > 0.995f
+                && !((preSel.geoId == Sketcher::GeoEnum::HAxis
+                      || preSel.geoId == Sketcher::GeoEnum::VAxis)
+                     && fabs(cosangle) < 0.1)) {

```

Comment:
I don't get this. Am I misunderstanding something, or is it a requirement, that

`fabs(cosangle) > 0.995f` and `!fabs(cosangle) < 0.1`? Which can be expressed solely as `fabs(cosangle) > 0.995f`?

Reply:
I do not know I have just refactored, ie copy pasted.
After some more analysis, I think you are correct. This if is equivalent to : 
`fabs(cosangle) > 1 && ((geoId != ... || geoId != ...) || fabs(cosangle) > 0.1)`
Which is equivalent to 
`fabs(cosangle) > 1`

Now the question is, should we just simplify or is there a bug generated by this?

Reply:
This was writen 11y ago by Abdullah : https://github.com/FreeCAD/FreeCAD/commit/501c40f0e662dce67561d0f90913cd3ff5ba1140#diff-22b1021f1c8f959269c4990f04b5bb09c9e3973a9ed1e7406d43bee147e2f031

So I guess it's probably safe to just remove it?